### PR TITLE
feat: do not try to upload files if the credentials a might wrong

### PR DESCRIPTION
**Pull request by cubasepp**
It took me some time to figure out what it's really wrong. 
Now if we have the option for multiple buckets it's might be better to see if the credentials are correct.

Unfortunately I was unable to test it correctly so I have alos no idea to build it local and test it and I also do not want to experiment to much with building. Maybe we can publish it as beta as well?
Or you give me a hint how to test it local.

Thanks.

### CODE REVIEW

**Comment by Jefiozie** [comment]
Thanks for your PR @cubasepp 🙏🏻, I'm going to have a look at it and will test it with AWS and provide a review.

File: builder/deploy/uploader.ts [hunk 1]
```diff
@@ -9,60 +9,92 @@ import {
...
           `❌  Looks like you are missing some configuration`,
         );
         return;
       }
+
+      const params: HeadBucketRequest = {
+        Bucket: this._bucket,
+      };
+
+      this._s3.headBucket(params)
```
Comment (Jefiozie) [inline builder/deploy/uploader.ts:52]:
```suggestion
await this._s3.headBucket(params)
```

File: builder/deploy/uploader.ts [hunk 2]
```diff
@@ -9,60 +9,92 @@ import {
...
+        .promise()
+        .then(() => {
+          return this.uploadFiles(files, filesPath);
+        })
+        .catch(() => {
+          this._context.logger.error(
+            `❌  Looks like your credentials are incorrect`,
+          );
```
Comment (Jefiozie) [inline builder/deploy/uploader.ts:60]: Could we log the error here. I think something like the following will be enough:
```
        .catch(error => {
          this._context.logger.error(`❌  The following error was found during the upload ${error}`);
```

**Comment by cubasepp** [comment]
Done. Thanks

**Comment by Jefiozie** [comment]
Thanks, will review it asap.

**Comment by Jefiozie** [comment]
Could you update your changes with the latest changes off the beta branch? Thanks!

**Comment by cubasepp** [comment]
@Jefiozie I rebased this branch with `master` and `beta` branch!

**Comment by cubasepp** [comment]
So I solved all open request I guess. How do you go forward? 🤔

**Comment by Jefiozie** [comment]
@all-contributors please add @cubasepp  for code

**Pull request merged**